pandas/types/dtypes.py

```diff
@@ -244,6 +244,119 @@ def __eq__(self, other):
         if isinstance(other, compat.string_types):
             return other == self.name
 
-        return isinstance(other, DatetimeTZDtype) and \
-            self.unit == other.unit and \
-            str(self.tz) == str(other.tz)
+        return (isinstance(other, DatetimeTZDtype) and
+                self.unit == other.unit and
+                str(self.tz) == str(other.tz))
+
+
+class PeriodDtypeType(type):
+    """
+    the type of PeriodDtype, this metaclass determines subclass ability
+    """
+    pass
+
+
+class PeriodDtype(ExtensionDtype):
+    __metaclass__ = PeriodDtypeType
+    """
+    A Period duck-typed class, suitable for holding a period with freq dtype.
+
+    THIS IS NOT A REAL NUMPY DTYPE, but essentially a sub-class of np.int64.
+    """
+    type = PeriodDtypeType
+    kind = 'O'
+    str = '|O08'
+    base = np.dtype('O')
+    num = 102
+    _metadata = ['freq']
+    _match = re.compile("period\[(?P<freq>.+)\]")
+
+    def __init__(self, freq, strict=False):
+        """
+        Parameters
+        ----------
+        freq : frequency
+        """
+
+        if isinstance(freq, PeriodDtype):
+            self.freq = freq.freq
+            return
+
+        if freq is None:
+            raise ValueError("could not construct PeriodDtype")
+
+        from pandas._period import Period
+        try:
+            freq = Period._maybe_convert_freq(freq)
+            if freq is not None:
+                self.freq = freq
+                return
+        except ValueError:
+            pass
+
+        self.freq = self._parse_dtype_strict(freq)
+
+    @classmethod
+    def _parse_dtype_strict(cls, freq):
+        try:
+            m = cls._match.search(freq)
+            if m is not None:
+                from pandas._period import Period
+                freq = m.groupdict()['freq']
+                freq = Period._maybe_convert_freq(freq)
+                if freq is not None:
+                    return freq
+        except:
+            pass
+        raise ValueError("could not construct PeriodDtype")
+
+    @classmethod
+    def construct_from_string(cls, string):
+        """
+        attempt to construct this type from a string, raise a TypeError
+        if its not possible
+        """
+        from pandas.tseries.offsets import DateOffset
+        if isinstance(string, (compat.string_types, DateOffset)):
+            # avoid tuple to be regarded as freq
+            try:
+                return cls(freq=string)
+            except ValueError:
+                pass
+        raise TypeError("could not construct PeriodDtype")
+
+    def __unicode__(self):
+        return "period[{freq}]".format(freq=self.freq.freqstr)
+
+    @property
+    def name(self):
```

Comment:
side thing. We should just fix the metaclass to put in all of these inheritet methods....but that's another PR!